.github/workflows/ci.yaml

```diff
@@ -103,6 +103,10 @@ jobs:
           RUSTC_BOOTSTRAP: 1
         run: target/${{ matrix.target }}/debug/rust-analyzer analysis-stats --with-deps $(rustc --print sysroot)/lib/rustlib/src/rust/library/std
 
+      - name: clippy
+        if: matrix.os == 'ubuntu-latest'
+        run: cargo clippy --all-targets -- -D warnings
```

Comment:
Does this enable *all clippy* warnings or specifically only the ones we list in the `Cargo.toml` files? We only want the latter as we want direct control of the lints.

Reply:
It does enable all warnings.

The approach I took is enabling all warnings, and then ignoring (aka. "allow"ing) all warnings we do not care about. Some warnings gonna be ignored forever, but some others are gonna be worked on (as happened in e.g. #16404).

Reply:
We could also only enable ("deny" in clippy jargon) the lints we care about, but I prefer the approach of considering all clippy lints, except we find them to be not fitting to our needs.

Reply:
Ah wait now I see. I'm confused as to why we pass `-D warnings` here, thats not a clippy lint group and clippy should check normal rustc warnings by default no? All the other groups that are relevant we have enabled in the lints table already.

Reply:
Good point. Gonna remove that.